### PR TITLE
wrong project platform is being used

### PR DESCRIPTION
**Pull request by elahehrashedi**
An extra env variable `Platform` is being sent by `collectDevBatVars` function and it is causing the "test" project build to fail.

### CODE REVIEW

File: src/kit.ts [hunk 1]
```diff
@@ -603,7 +603,7 @@ const MSVC_ENVIRONMENT_VARIABLES = [
   'LIBPATH',
   'NETFXSDKDir',
   'Path',
-  'Platform',
+  //'Platform',
```
Comment (bobbrow) [inline src/kit.ts:606]: I don't actually see Platform ever set in any of the Developer Command prompts I've opened (VS 2017-VS 2022).  We should just delete this line.

cc: @lygstate

Comment (lygstate) [reply src/kit.ts:606]: Platform  is set by VS2019, maybe we need a better way to detecting if Platform are set?

**Comment by lygstate** [comment]
```
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.10.0
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x86_x64'

C:\Program Files (x86)\Microsoft Visual Studio\2019\Community>echo %Platform%
x64

C:\Program Files (x86)\Microsoft Visual Studio\2019\Community>
```

**Comment by elahehrashedi** [comment]
test if we need to set platform when host and target kits are different? 
test building cpptools using just 64 without setting platform (setting host and target the same)

**Comment by lygstate** [comment]
I have no object to remove `Platform`, I am just telling where the `Platform` comes from

**Comment by bobbrow** [comment]
Thanks @lygstate, we just wanted to make sure removing it does the right thing for us, so these are a few notes @elahehrashedi wrote while we were discussing it this morning.  We were also hoping it doesn't break your fix for msvc2015 detection.

**Pull request merged**